### PR TITLE
Fragments `componentGroup` can be changed via scenario properties

### PR DESCRIPTION
**Pull request by wrzontek**
## Describe your changes

Screenshot from 2024-07-24 11-43-45

Screenshot from 2024-07-24 11-43-49

Screenshot from 2024-07-24 11-47-46

Screenshot from 2024-07-24 11-46-05

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Comment by wrzontek** [comment]
CI green aside from flaky tests.

File: designer/server/src/main/scala/db/migration/V1_054__FragmentAddComponentGroupProperty.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import io.circe._
+import pl.touk.nussknacker.ui.db.migration.ProcessJsonMigration
+
+trait V1_054__FragmentAddComponentGroupPropertyDefinition extends ProcessJsonMigration {
+
+  override def updateProcessJson(jsonProcess: Json): Option[Json] =
+    V1_054__FragmentAddComponentGroupProperty.migrateMetadata(jsonProcess)
+}
+
+object V1_054__FragmentAddComponentGroupProperty {
```
Comment (arkadius) [inline designer/server/src/main/scala/db/migration/V1_054__FragmentAddComponentGroupProperty.scala:12]: Is it needed? Can't we use "in-flight migration approach" I mean `FragmentSpecificData.apply` would have somewhere getOrElse (default group). BTW it is safer to not store the group until it won't be changed. We don't have separate id / name for component groups. So once we store this "fragments" group in some scenario, changing it to e.g. "Fragments" will be painfull.

Comment (wrzontek) [reply designer/server/src/main/scala/db/migration/V1_054__FragmentAddComponentGroupProperty.scala:12]: I considered "in-flight", but that means that when you edit a fragment from before this change, and go into it's properties, the "Component group" field will be empty

if i understand correctly, it's taken straight from the process json which is taken straight from the database without altering, so to set a default on BE would require modyfing it between the DB and FE when fetching process, which currently isn't done

that's why i chose migration as the lesser evil

Comment (wrzontek) [reply designer/server/src/main/scala/db/migration/V1_054__FragmentAddComponentGroupProperty.scala:12]: image
 :D

Comment (arkadius) [reply designer/server/src/main/scala/db/migration/V1_054__FragmentAddComponentGroupProperty.scala:12]: Ehh, you're right

**Review by arkadius — APPROVED** [review]
LGTM

**Pull request merged**